src/support/import-supervisor.js

```diff
@@ -176,19 +177,36 @@ function ImportSupervisor(services, config) {
    * @param {string} importScript - Optional custom Base64 encoded import script.
    * @returns {Promise<ImportJob>}
    */
-  async function startNewJob(urls, importApiKey, options, importScript, initiatedBy) {
+  async function startNewJob(
+    urls,
+    importApiKey,
+    options,
+    importScript,
+    initiatedBy,
+    customHeaders,
+  ) {
     // Determine if there is a free import queue
     const importQueueId = await getAvailableImportQueue(importApiKey);
 
     // Hash the API Key to ensure it is not stored in plain text
     const hashedApiKey = hashWithSHA256(importApiKey);
 
+    let updatedOptions = { ...options };
+
+    if (importScript) {
+      updatedOptions = { ...updatedOptions, isCustomJavascriptProvided: true };
```

Comment:
Suggestion for readability:
```suggestion
      updatedOptions = { ...updatedOptions, hasCustomImportJs: true };
```